esmf_regrid/schemes.py

```diff
@@ -457,37 +458,35 @@ def copy_coords(src_coords, add_method):
 
 
 def _make_gridinfo(cube, method, resolution, mask):
+    method = check_method(method)
     if resolution is not None:
         if not (isinstance(resolution, int) and resolution > 0):
             raise ValueError("resolution must be a positive integer.")
-        if method != "conservative":
+        if method != Constants.Method.CONSERVATIVE:
             raise ValueError("resolution can only be set for conservative regridding.")
-    if method == "conservative":
+    if method == Constants.Method.CONSERVATIVE:
         center = False
-    elif method in ("bilinear", "nearest"):
+    elif method in (Constants.Method.NEAREST, Constants.Method.BILINEAR):
         center = True
-    else:
-        raise NotImplementedError(
-            f"method must be either 'bilinear', 'conservative' or 'nearest', got '{method}'."
-        )
     return _cube_to_GridInfo(cube, center=center, resolution=resolution, mask=mask)
 
 
 def _make_meshinfo(cube_or_mesh, method, mask, src_or_tgt, location=None):
+    method = check_method(method)
     if isinstance(cube_or_mesh, Mesh):
         mesh = cube_or_mesh
     else:
         mesh = cube_or_mesh.mesh
         location = cube_or_mesh.location
     if mesh is None:
         raise ValueError(f"The {src_or_tgt} cube is not defined on a mesh.")
-    if method == "conservative":
+    if method == Constants.Method.CONSERVATIVE:
         if location != "face":
             raise ValueError(
                 f"Conservative regridding requires a {src_or_tgt} cube located on "
```

Comment:
For the sake of consistency, it's worth considering changing "Conservative" to "`{method}`". I think it's also worth either changing `{method}` for `{method.name}` for the sake of clarity and readability or else rewriting this error message so that the formatting of `{method}` reads less awkwardly (e.g. "regridding with the method {method} requires...").